braidpool-primitives/src/braid/mod.rs

```diff
@@ -42,20 +45,41 @@ impl Braid {
         Braid {
             beads: genesis_beads.clone(),
             tips: genesis_beads.clone(),
-            cohorts: vec![Cohort(genesis_beads)],
+            cohorts: vec![Cohort(genesis_beads.clone())],
             orphan_beads: Vec::new(),
             loaded_beads_in_memory: HashMap::new(),
+            genesis_beads: genesis_beads,
+        }
+    }
+
+    pub fn load_beads_in_memory(&mut self, beads: Vec<Bead>) {
+        for bead in beads {
+            let serialized_curr_bead_str_result = serde_json::to_string(&bead);
+            let serialized_curr_bead_str = match serialized_curr_bead_str_result {
+                Ok(serialized_str) => serialized_str,
+                Err(error) => {
+                    panic!(
+                        "An error occrrred while seralizing the bead string while loading beads in memory {:?} ",
+                        error
+                    );
+                }
+            };
+            let mut serialized_bytes = [0u8; 32];
+            hex::decode_to_slice(serialized_curr_bead_str, &mut serialized_bytes).unwrap();
+            let bead_hash = BlockHash::from_byte_array(serialized_bytes);
+            self.loaded_beads_in_memory.insert(bead_hash, bead);
         }
     }
 
     pub fn generate_from_previous_dag(previous_dag_braid: Braid) -> Self {
```

Comment:
This is effectively a copy constructor...

How do you see this being used? I'm not sure we'll ever want to copy a braid...